src/pages/index.js

```diff
@@ -1,130 +1,223 @@
+      console.error("Caught an error: ", error.message);
+    });
+
   //functions
+  const deletePopup = new ConfirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+    const items = [{ name: name, description: description }];
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+        deletePopup.open();
+        deletePopup.setDefaultSaveButton();
+        deletePopup.setSubmitAction(() => {
+          api
+            .deleteCard(card.id)
+            .then(() => {
+              card.deleteCard();
+            })
+            .then(() => {
+              deletePopup.updateSaveButton();
+            })
+            .then(() => {
+              deletePopup.close();
+            })
+            .catch((error) => {
+              console.error("Caught an error: ", error.message);
+            });
+        });
+      },
+      toggleLikeButton,
+      initialCardsInfo,
+      id
+    );
+    cardsSection.items = items;
+    cardsSection.addItem(modelCard.generateCard());
+  }
 
   function createCard(name, description, template) {
-    const modelCard = new Card(name, description, template, imageClickHandler);
-    cardsSection.addItem(modelCard.addCard());
+    return api
+      .addCardInfo(name, description, (name, link, id) => {
+        addCard(name, link, template, id);
+      })
+      .then(
+        addPopup.close()
+      )
+      .catch((error) => {
+        console.error("Caught an error: ", error.message);
+      })
+      .finally(() => {
+        addPopup.submitBtn.textContent = "Saving...";
+      });
   }
 
-  function handleProfileEditSubmit({ name, description }) {
-    userInfo.setUserInfo({ name: name, about: description });
-    editPopup.close();
+  function handleProfileEditSubmit(inputElements) {
+    api
+      .setUserInfo(inputElements.name, inputElements.description)
+      .then((result) => {
+        user.setUserInfo({ name: result.name, about: result.about });
+      })
+      .then(() => {
+        editPopup.close();
+      })
+      .catch((error) => {
+        console.error("Caught an error: ", error.message);
+      })
+      .finally(() => {
+        editPopup.submitBtn.textContent = "Saving...";
+      });
   }
 
   function handleProfileAddSubmit({ name, description }) {
-    createCard(name, description, cardTemplate);
-    addPopup.close();
+    createCard(name, description, Constants.cardTemplate);
   }
 
   //event listeners
 
-  profileEditButton.addEventListener("click", () => {
+  Constants.profileEditButton.addEventListener("click", () => {
+    const data = user.getUserInfo();
     editPopup.setPreviewedValues({
-      name: profileName.innerText,
-      description: profileDescription.innerText,
+      name: data.name,
+      description: data.about,
     });
     editPopup.open();
-  });
+    editPopup.submitBtn.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.